collaborative-circom/src/lib.rs

```diff
@@ -97,31 +97,395 @@ impl std::fmt::Display for MPCProtocol {
+    /// Network config
+    pub network: NetworkConfig,
+}
+
+/// Cli arguments for `verify`
+#[derive(Debug, Serialize, Args)]
+pub struct VerifyCli {
+    /// The proof system to be used
+    #[arg(value_enum)]
+    pub proof_system: ProofSystem,
+    /// The path to the config file
+    #[arg(long)]
+    #[serde(skip_serializing_if = "::std::option::Option::is_none")]
+    pub config: Option<PathBuf>,
+    /// The path to the proof file
+    #[arg(long)]
+    #[serde(skip_serializing_if = "::std::option::Option::is_none")]
+    pub proof: Option<PathBuf>,
+    /// The pairing friendly curve to be used
+    #[arg(long, value_enum)]
+    #[serde(skip_serializing_if = "::std::option::Option::is_none")]
+    pub curve: Option<MPCCurve>,
+    /// The path to the verification key file
+    #[arg(long)]
+    #[serde(skip_serializing_if = "::std::option::Option::is_none")]
+    pub vk: Option<PathBuf>,
+    /// The path to the public input JSON file
+    #[arg(long)]
+    #[serde(skip_serializing_if = "::std::option::Option::is_none")]
+    pub public_input: Option<PathBuf>,
+}
+
+/// Config for `verify`
+#[derive(Debug, Deserialize)]
+pub struct VerifyConfig {
+    /// The proof system to be used
+    pub proof_system: ProofSystem,
+    /// The path to the proof file
+    pub proof: PathBuf,
+    /// The pairing friendly curve to be used
+    pub curve: MPCCurve,
+    /// The path to the verification key file
+    pub vk: PathBuf,
+    /// The path to the public input JSON file
+    pub public_input: PathBuf,
+}
+
+/// Prefix for config env variables
+pub const CONFIG_ENV_PREFIX: &str = "COCIRCOM_";
+
+/// Error type for config parsing and merging
+#[derive(thiserror::Error, Debug)]
+#[error(transparent)]
+pub struct ConfigError(#[from] figment::error::Error);
+
+macro_rules! impl_try_from {
+    ($cli: ty, $config: ty) => {
+        impl TryFrom<$cli> for $config {
+            type Error = ConfigError;
+            fn try_from(cli: $cli) -> Result<Self, Self::Error> {
+                if let Some(path) = &cli.config {
+                    Ok(Figment::new()
+                        .merge(Toml::file(path))
+                        .merge(Env::prefixed(CONFIG_ENV_PREFIX))
+                        .merge(Serialized::defaults(cli))
+                        .extract()?)
+                } else {
+                    Ok(Figment::new()
+                        .merge(Env::prefixed(CONFIG_ENV_PREFIX))
+                        .merge(Serialized::defaults(cli))
+                        .extract()?)
+                }
+            }
+        }
+    };
+}
+
+impl_try_from!(SplitInputCli, SplitInputConfig);
+impl_try_from!(SplitWitnessCli, SplitWitnessConfig);
+impl_try_from!(MergeInputSharesCli, MergeInputSharesConfig);
+impl_try_from!(GenerateWitnessCli, GenerateWitnessConfig);
+impl_try_from!(TranslateWitnessCli, TranslateWitnessConfig);
+impl_try_from!(GenerateProofCli, GenerateProofConfig);
+impl_try_from!(VerifyCli, VerifyConfig);
```

Comment:
I don't think `try_from` should have these side-effects? (loading stuff from env/file, etc)
Maybe just make this a method on the target config struct, also via macro